face/unix-stream-transport.go

```diff
@@ -117,20 +117,32 @@ func (t *UnixStreamTransport) runReceive() {
 		}
 
 		// Determine whether valid packet received
-		for startPos > 0 {
-			_, _, tlvSize, err := tlv.DecodeTypeLength(recvBuf)
+		tlvPos := 0
+		for {
+			if tlvPos >= startPos {
+				startPos = 0
+				break
+			}
+
+			_, _, tlvSize, err := tlv.DecodeTypeLength(recvBuf[tlvPos:])
 			if err != nil {
 				core.LogInfo(t, "Unable to process received packet: ", err)
 				startPos = 0
 				break
-			} else if startPos >= tlvSize {
+			} else if startPos >= tlvPos+tlvSize {
 				// Packet was successfully received, send up to link service
-				t.linkService.handleIncomingFrame(recvBuf[:tlvSize])
-				startPos -= tlvSize
-				if startPos > 0 {
-					copy(recvBuf, recvBuf[tlvSize:])
-				}
+				t.linkService.handleIncomingFrame(recvBuf[tlvPos : tlvPos+tlvSize])
+				tlvPos += tlvSize
 			} else {
+				if tlvPos > 0 {
```

Comment:
Since this is a stream transport, let's assume that fragmented packets will be received.
Suppose the transport just started and we received the first 10B of a 20B packet. Then we have:

- `startPos = 10`
- `tlvPos = 0`
- `tlvSize = 20`

So the program will get to this branch and drop the packet because `tlvPos == 0`.
Is this supposed behavior?

Reply:
OK I see.